ldap/servers/plugins/replication/repl5_replica.c

```diff
@@ -829,6 +829,37 @@ replica_set_ruv(Replica *r, RUV *ruv)
     replica_unlock(r->repl_lock);
 }
 
+/*
+ * Check if replica generation is the same than the remote ruv one
+ */
+int
+replica_check_generation(Replica *r, const RUV *remote_ruv)
+{
+    int return_value;
+
+    char *local_gen = NULL;
+    char *remote_gen = ruv_get_replica_generation(remote_ruv);
+    Object *local_ruv_obj;
+    RUV *local_ruv;
+
+    PR_ASSERT(NULL != r);
+    local_ruv_obj = replica_get_ruv(r);
+    if (NULL != local_ruv_obj) {
+        local_ruv = (RUV *)object_get_data(local_ruv_obj);
+        PR_ASSERT(local_ruv);
+        local_gen = ruv_get_replica_generation(local_ruv);
+        object_release(local_ruv_obj);
+    }
+    if (NULL == remote_gen || NULL == local_gen || strcmp(remote_gen, local_gen) != 0) {
+        return_value = PR_FALSE;
+    } else {
+        return_value = PR_TRUE;
+    }
+    slapi_ch_free((void **)&remote_gen);
+    slapi_ch_free((void **)&local_gen);
```

Comment:
This is very minor, but we try and use slapi_ch_free_string() when freeing char pointers (instead of slapi_ch_free()).  I realize you were just moving the code around, but it's still nice to do this clean up whenever we can.

Reply:
fixed in latest force push